test.cu

```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+#include <cuda_runtime.h>
+
+__global__ void testDependency(int *start, int *results, int n)
+{
+  int index = threadIdx.x + blockIdx.x * blockDim.x;
+
+  int dependency_var = 0;
+
+  // # pragma unroll
+  for (int i = index; i < n; ++i)
+  {
+    // if (start[i] < n && i == start[i])
+    // {
+      dependency_var = start[i];
+    // }
+  }
+  results[index] = dependency_var;
+}
+
+int main()
+{
+  unsigned n = 1024*1024;
+  int results[n];
+
+  int *d_results;
+  cudaMalloc(&d_results, n * sizeof(int));
+
+  int *start, *d_start;
+  cudaMalloc(&d_start, n * sizeof(int));
+  cudaMallocHost(&start, n * sizeof(int));
```

Comment:
start and d_start are allocated but not freed

Reply:
removed